Stack/Opc.Ua.Core/Types/Encoders/BinaryDecoder.cs

```diff
@@ -1530,7 +1530,10 @@ public object ReadArray(string fieldName, int valueRank, BuiltInType builtInType
                                 m_context.MaxArrayLength);
                         }
 
-                        length *= dimensions[ii];
+                        checked
```

Comment:
Same comment, OverflowException feeld like errors from the stack, I suggest using ulong for length and throw an dedicated ArgumentException once it is getting to big

Reply:
Same response as to the first comment :)